### PR TITLE
[MGDOBR-753] - Add Productization profile to exclude non-required dep…

### PR DESCRIPTION
**Pull request by vaibhavjainwiz**
…endencies

Jira issue - https://issues.redhat.com/browse/MGDOBR-754

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] The layers in the `kustomize` folder have been updated accordingly.
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -45,12 +45,33 @@
...
+  <profiles>
+    <profile>
+      <id>default</id>
+      <activation>
+        <property>
+          <name>default</name>
+        </property>
+        <activeByDefault>true</activeByDefault>
+      </activation>
+      <modules>
```
Comment (r00ta) [inline pom.xml:62]: Can you add a comment here like `add here all the modules that should not be in the productized profile`?

Comment (vaibhavjainwiz) [reply pom.xml:63]: done

File: manager/pom.xml [hunk 2]
```diff
@@ -156,11 +156,6 @@
       <artifactId>logger-java-log4j</artifactId>
       <scope>test</scope>
     </dependency>
-    <dependency>
-      <groupId>io.quarkus</groupId>
-      <artifactId>quarkus-jacoco</artifactId>
```
Comment (danielezonca) [inline manager/pom.xml:161]: Is this an unused dependency?

File: build-parent/pom.xml [hunk 3]
```diff
@@ -495,6 +502,14 @@
         <formatter.goal>validate</formatter.goal>
         <impsort.goal>check</impsort.goal>
       </properties>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.jacoco</groupId>
+            <artifactId>jacoco-maven-plugin</artifactId>
```
Comment (danielezonca) [inline build-parent/pom.xml:509]: Are you sure it is necessary to declare this dependency also here? As far as I see it seems to be used only by the `measure-code-coverage` profile

**Review by danielezonca — APPROVED** [review]
LGTM, two minor comments

File: manager/pom.xml (hunk 2 above)
Comment (vaibhavjainwiz) [reply manager/pom.xml:161]: I check the `quarkus-jacoco` documentation, it mentioned that developer could use either plugin or dependency.
https://quarkus.io/guides/tests-with-coverage

`jacoco-maven-plugin` plugin is included with `measure-code-coverage` profile.

**Review by r00ta — APPROVED** [review]

**Comment by manstis** [comment]
@sutaakar @jomarko @vaibhavjainwiz FYI. This might have broken our builds. Keep an eye open on PR statuses!

**Comment by vaibhavjainwiz** [comment]
@manstis 
Its very strange. I didn't make any such change which could cause this issue.
On which step you got this error mean at time of compilation or running?
For this PR integration test are failing because of MC issue whose discussion is still going on.

**Comment by manstis** [comment]
@vaibhavjainwiz I just rebased my local with `main` (including this commit) and did a `mvn clean install`.

**Pull request merged**